ci(napi/oxlint): run tests with `coverage` profile

overlookmotel opened the pull request:
The bug fixed in #12299 went unnoticed originally because we build the `napi/oxlint2` crate for tests in release mode, with debug assertions disabled.

Instead, build with `--profile coverage` which enables debug assertions. It's also faster to compile.

overlookmotel commented:
* **#12308** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12308?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12308?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12308).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 reviewed (commented):
should we do this for all napi packages?

overlookmotel committed:
ci(napi/oxlint): run tests with `coverage` profile (#12308)

The bug fixed in #12299 went unnoticed originally because we build the `napi/oxlint2` crate for tests in release mode, with debug assertions disabled.

Instead, build with `--profile coverage` which enables debug assertions. It's also faster to compile.

overlookmotel commented:
> should we do this for all napi packages?

Not sure.

`oxc-parser` tests already take ~9 seconds to run. It'd be ideal not to slow them down further if we can avoid it, and the compile time is pretty good already. But maybe enabling debug assertions and `-O 2` optimizations wouldn't affect runtime much?

The rest of the napi packages I know nothing about!
